### PR TITLE
Refactor plugin command + add option select

### PR DESCRIPTION
**Pull request by jellz**
TL;DR This PR refactors the plugin command to improve the code, make the response message more user-friendly and add caching to improve speed and efficiency. The PR also fixes problems with resolving plugins.

- **Caching**
The current code fetches the entire plugin list every time the command is run and uses that data to find the plugin. Since this PR uses the Minehut API library, it benefits from plugin caching.

- **Typings**
The current code uses the `any` type for plugins which isn't good for type-safety and maintainability. This PR changes it to use the existing Minehut API library which includes typings for Minehut plugins. This PR also removes some unnecessary explicit typings.

- **Name searching**
Currently, the command only looks for exact (case-insensitive) matches of ID or plugin name. This is not ideal, especially when looking up plugins with special prefixes (e.g. `Skript Addon: TuSKe`). This PR [revisits the searching](https://i.imgur.com/b3uTjSY.gif), to look for exact (case-insensitive) name/ID matches first, and then look for name inclusions of the query. If the query returns multiple plugins, the bot will [let the user choose the plugin from the list](https://i.imgur.com/wihCxIe.mp4).

- **Response**
PR uses the plugin's extended description, whereas the old code only used its short description (much less descriptive). PR also omits some less important embed fields from the old code to improve UX, and the embed colour will change based on if the plugin is private or not ([red if private](https://i.imgur.com/7QR3Jag.png)). PR also renames Created to Added, to reflect that the plugin was not created on that date, but rather added to Minehut.

- **Error prone**
The old code fails if the queried plugin is missing a link in its extended description, causing the bot to just [hang](https://i.imgur.com/Oeyd6J1.png) -- that issue is [fixed here](https://i.imgur.com/k2Djhl1.png).

### CODE REVIEW

**Review by Sniped — APPROVED** [review]
I like the use of the Minehut API library here, great PR!

**Comment by theminecoder** [comment]
So taking a look at the minehut lib updates (which tbh should probably be moved into the minehut org, but that's a matter for another time) there seems to be no way for this cache to expire, which is going to be needed as plugin updates happen often and the bot can be up for some time before updates make it restart. So before this is merged that will need to happen.

**Review by theminecoder — CHANGES_REQUESTED** [review]
When the changes are made please request a re-review as I don't have updates on the other repo set to notify me

**Comment by jellz** [comment]
> So taking a look at the minehut lib updates (which tbh should probably be moved into the minehut org, but that's a matter for another time) there seems to be no way for this cache to expire, which is going to be needed as plugin updates happen often and the bot can be up for some time before updates make it restart. So before this is merged that will need to happen.

I thought about that. I figured the bot might restart at least once per week but I suppose it could not. I'll add expiration to the cache tomorrow.

**Comment by jellz** [comment]
Added plugin cache expiry of 30 minutes.

**Review by theminecoder — APPROVED** [review]
